### PR TITLE
drop support for legacy (v0) projects

### PR DESCRIPTION
**Pull request by droot**
In this PR, I have tried to avoid deleting code related to `v0 projects,  instead I am just dropping support for `v0` from the commands. I will be deleting the code related to `v0` projects in upcoming PRs and then followed by restructuring the code to make it easy to understand.

This is a first in the series of PRs for cleaning up
the technical debt.

### CODE REVIEW

File: cmd/kubebuilder/main.go [hunk 1]
```diff
@@ -49,14 +48,15 @@ func main() {
...
+	// add version command
 	version.AddVersion(cmd)
 
-	if util.IsNewVersion() || util.IsProjectNotInitialized() {
-		v1.AddCmds(cmd)
-	} else {
-		v0.AddCmds(cmd)
-	}
+	// add all commands corresponding to v1 project
+	v1.AddCmds(cmd)
```
Comment (Liujingfang1) [inline cmd/kubebuilder/main.go]: This v1 package can be removed after v0 cleanup.

File: .travis.yml [hunk 2]
```diff
@@ -22,8 +22,6 @@ install:
   -
 
 script:
-- TRACE=1 ./testv0.sh
```
Comment (Liujingfang1) [inline .travis.yml]: Nice. This will reduce the ci time quite a lot.

File: cmd/kubebuilder/initproject/project.go [hunk 3]
```diff
@@ -44,6 +43,12 @@ type projectOptions struct {
 }
 
 func (o *projectOptions) RunInit() {
+	checkGoVersion()
```
Comment (Liujingfang1) [inline cmd/kubebuilder/initproject/project.go]: Does it make sense to combine this file with `init.go`?

File: cmd/kubebuilder/main.go (hunk 1 above)
Comment (droot) [reply cmd/kubebuilder/main.go]: Yes. I am keeping code deletion for next PR.

File: cmd/kubebuilder/initproject/project.go (hunk 3 above)
Comment (droot) [reply cmd/kubebuilder/initproject/project.go]: Good point, just did that.

**Pull request merged**